lib/msf_autoload.rb

```diff
@@ -14,274 +14,319 @@
+      'meterpreter_mips64_linux' => 'Meterpreter_mips64_Linux',
+      'meterpreter_x86_linux' => 'Meterpreter_x86_Linux',
+      'meterpreter_mipsbe_linux' => 'Meterpreter_mipsbe_Linux',
+      'meterpreter_aarch64_apple_ios' => 'Meterpreter_aarch64_Apple_iOS',
+      'meterpreter_x64_osx' => 'Meterpreter_x64_OSX',
+      'meterpreter_ppc_linux' => 'Meterpreter_ppc_Linux',
+      'meterpreter_x64_win' => 'Meterpreter_x64_Win',
+      'meterpreter_php' => 'Meterpreter_Php_Php',
+      'meterpreter_multi' => 'Meterpreter_Multi',
+      'hwbridge' => 'HWBridge',
+      'vncinject_options' => 'VncInjectOptions',
+      'vncinject' => 'VncInject',
+      'json_hash_file' => 'JSONHashFile',
+      'ndr' => 'NDR',
+      'ci_document' => 'CIDocument',
+      'fusionvm_document' => 'FusionVMDocument',
+      'group_policy_preferences' => 'GPP',
+      'ip360_aspl_xml' => 'IP360ASPLXMLStreamParser',
+      'ip360_xml' => 'IP360XMLStreamParser',
+      'nessus_xml' => 'NessusXMLStreamParser',
+      'netsparker_xml' => 'NetSparkerXMLStreamParser',
+      'nexpose_xml' => 'NexposeXMLStreamParser',
+      'nmap_xml' => 'NmapXMLStreamParser',
+      'openvas_document' => 'OpenVASDocument',
+      'retina_xml' => 'RetinaXMLStreamParser',
+      'graphml' => 'GraphML',
+      'apple_backup_manifestdb' => 'AppleBackupManifestDB',
+      'winscp' => 'WinSCP',
+      'acpp' => 'ACPP',
+      'tftp' => 'TFTP',
+      'ipmi' => 'IPMI',
+      'channel_auth_reply' => 'Channel_Auth_Reply',
+      'open_session_reply' => 'Open_Session_Reply',
+      'rakp2' => 'RAKP2',
+      'pjl' => 'PJL',
+      'dhcp' => 'DHCP',
+      'addp' => 'ADDP',
+      'rfb' => 'RFB',
+      'io' => 'IO',
+      'ntfs' => 'NTFS',
+      'bitlocker' => 'BITLOCKER',
+      'adb' => 'ADB',
+      'drda' => 'DRDA',
+      'tlv' => 'TLV',
+      'svcctl' => 'SVCCTL',
+      'wdscp' => 'WDSCP',
+      'appapi' => 'AppApi',
+      'uds_errors' => 'UDSErrors'
+    }
+  end
 
-loader.setup # ready!
+  def config_paths
+    [
+      { path: "#{__dir__}/msf/", namespace: Msf },
+      { path: "#{__dir__}/rex/", namespace: Rex },
+      { path: "#{__dir__}/../app/validators/" },
+    ]
+  end
+
+  # Enables :prepend to inject  existing loader
+  def create_loader
+    Zeitwerk::Loader.new
+  end
+
+  # Enables :prepend to override the configuration items pass to the loader
+  def configure(loader)
+    config_paths.each do |entry|
+      if entry[:namespace]
+        loader.push_dir(entry[:path], namespace: entry[:namespace])
+      else
+        loader.push_dir(entry[:path])
+      end
+    end
+    loader.ignore(ignore_list)
+    loader.collapse(collapse_list)
+    loader.inflector = TempInflector.new
+    loader.inflector.inflect(custom_inflections)
+  end
+
+  # Enables :prepend to suppress the loader finalization
+  def finalize_loader(loader)
+    loader.setup # ready!
+    
+    # global autoload of common gems
+    autoload :Faker, 'faker'
+    autoload :BinData, 'bindata'
```

Comment:
This will need to exist outside of the class otherwise it wouldn't be accessible"
```ruby
$ cat main.rb 
class Foo
    autoload :FakeGem, './fake_gem'
end


puts FakeGem
```

Gives an error:
```ruby
$  ruby main.rb 
Traceback (most recent call last):
main.rb:6:in `<main>': uninitialized constant FakeGem (NameError)
```

Whilst extracting it works:
```ruby
class Foo
end

autoload :FakeGem, './fake_gem'

puts FakeGem
```